ley.config.mjs

```diff
@@ -0,0 +1,42 @@
+import fs from 'node:fs';
+import dotenv from 'dotenv';
+
+export function setEnvironmentVariables() {
+  if (process.env.NODE_ENV === 'production' || process.env.CI) {
+    // Set standard environment variables for Postgres.js from Vercel environment variables
+    if (process.env.POSTGRES_URL) {
+      process.env.PGHOST = process.env.POSTGRES_HOST;
+      process.env.PGDATABASE = process.env.POSTGRES_DATABASE;
+      process.env.PGUSERNAME = process.env.POSTGRES_USER;
+      process.env.PGPASSWORD = process.env.POSTGRES_PASSWORD;
+    }
+    return;
+  }
+
+  // Replacement for unmaintained dotenv-safe package
+  // https://github.com/rolodato/dotenv-safe/issues/128#issuecomment-1383176751
+  //
+  // TODO: Remove this and switch to dotenv/safe if this proposal gets implemented:
+  // https://github.com/motdotla/dotenv/issues/709
+  dotenv.config();
+
+  const unconfiguredEnvVars = Object.keys(
+    dotenv.parse(fs.readFileSync('./.env.example')),
+  ).filter((exampleKey) => !process.env[exampleKey]);
+
+  if (unconfiguredEnvVars.length > 0) {
+    throw new Error(
+      `.env.example environment ${
+        unconfiguredEnvVars.length > 1 ? 'variables' : 'variable'
+      } ${unconfiguredEnvVars.join(', ')} not configured in .env file`,
+    );
+  }
+}
+
+setEnvironmentVariables();
```

Comment:
interesting, just realized that the side effect of this is that when we import the file, it also runs this function... maybe not a problem, but we should decide what we want to do with this

Reply:
ok so you mean this function is going to run twice in the database connect file? 

still, I can't see why this may be an issue. But I am also open to using a different pattern then

Reply:
yeah we shouldn't run it twice - see if you can come up with a proposal or 2

Reply:
Options:

1. Moving this to utils/config.mjs as we had thought on Friday
2. Separate the logic so only dotenv is being called on import 

```js
import fs from 'node:fs';
import dotenv from 'dotenv';

function envConfig() {

  dotenv.config();
  ... rest of the file check
}

export function setEnvironmentVariables() {
  if (process.env.NODE_ENV === 'production' || process.env.CI) {
    // Set standard environment variables for Postgres.js from Vercel environment variables
    if (process.env.POSTGRES_URL) {
   .... env update
    }
    return;
  }
}

envConfig();

const options = {
  ssl: Boolean(process.env.POSTGRES_URL),
};

export default options;
```

Reply:
hmm, yeah option 2 probably wouldn't be great, makes the code hard to understand

other options:

3. change the import in `database/connect.ts` to `import '../ley.config.mjs'` without any named import, this will cause the function to run as a side effect
4. add a condition around the `setEnvironmentVariables()` in `ley.config.mjs` to prevent running a 2nd time - check for either:
   1. an environment variable that exists when Next.js is started
   2. an environment that was set by the `setEnvironmentVariables()` function
5. add some kind of boolean variable that checks whether the `setEnvironmentVariables()` function has already run, and then avoids running it a second time

but probably your option 1 seems best

Reply:
2a1f618